client/question/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -313,6 +313,28 @@ tie.factory('PythonCodePreprocessorService', [
       return VALID_WHITESPACE_REGEX.test(c);
     };
 
+    /**
+    * Creates and sets the string separator for user print stdOut.
+    *
+    * @returns {string}
+    * @private
+    */
+    var _generateStringSeparator = function() {
+      var chars = 'AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz';
+      var separator = '';
+      for (var i = 0; i < SEPARATOR_LENGTH; i++) {
+        var index = Math.floor(Math.random() * chars.length);
+        separator += chars.charAt(index);
+      }
+      StdOutSeparatorService.setSeparator(separator);
```

Comment:
Per notes above, it might not be worth having a StdOutSeparatorService.

Instead, you could pass the separator string along with the preprocessed code to the code runner service. The latter can then use it to split the output returned from the code run into per-test-case output.

Reply:
Thanks for the suggestion! Removed StdOutSeparatorService and instead created PreprocessedCode to handle both attributes.